logger.go

```diff
@@ -130,15 +130,25 @@ func (v *raftFormatter) formatDefault(writer io.Writer, level int, msg string, a
 		writer.Write([]byte(fmt.Sprintf("(%s) ", v.tag)))
 	}
 
-	// This is a little jank because the depth depends on the internals of
-	// the logxi library, but it's only used during unit tests.
+	// This is a little jank, but it's only used during unit tests.
 	if v.withLine {
-		_, file, line, ok := runtime.Caller(4)
-		if !ok {
-			file = "???"
-			line = 0
+		file := "???"
+		line := 0
+		depth := 0
+		for {
```

Comment:
I thought about doing this and hoped we didn't have to, but didn't catch the `Error()` case. Should be fine for tests, I agree.